src/System.Diagnostics.Process/src/System/Diagnostics/Process.NonUap.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace System.Diagnostics
+{
+    public partial class Process : IDisposable
+    {
+        public void Kill(bool entireProcessTree)
+        {
+            if (!entireProcessTree)
+            {
+                Kill();
+            }
+            else
+            {
+                EnsureState(State.Associated | State.IsLocal | State.HaveNonExitedId);
+
+                if (IsSelfOrDescendantOf(GetCurrentProcess()))
+                    throw new InvalidOperationException(SR.KillEntireProcessTree_DisallowedBecauseTreeContainsCallingProcess);
+
+                IEnumerable<Exception> result = KillTree();
+
+                if (result.Any())
+                    throw new AggregateException(SR.KillEntireProcessTree_TerminationIncomplete, result);
+            }
+        }
+
+        private bool IsSelfOrDescendantOf(Process processOfInterest)
+        {
+            if (SafePredicateTest(() => Equals(processOfInterest)))
+                return true;
+
+            Process[] allProcesses = GetProcesses();
+
+            try
+            {
+                var descendantProcesses = new Queue<Process>();
+                Process current = this;
+
+                do
+                {
+                    foreach (Process candidate in current.GetChildProcesses(allProcesses))
+                    {
+                        if (SafePredicateTest(() => processOfInterest.Equals(candidate)))
+                            return true;
+
+                        descendantProcesses.Enqueue(candidate);
+                    }
+                } while (descendantProcesses.TryDequeue(out current));
```

Comment:
nit: repo convention see that the `while` should be on its own line below the `}`. https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Buffers/ConfigurableArrayPool.cs#L86-L99

Reply:
I did a regex and there's 481 on the same line and 600 on separate lines. I call that a wash 😃